lib/adminSession.js

```diff
@@ -0,0 +1,94 @@
+import { googleCredentials, oauth2URI } from '../config/keys'
+import camelize from 'camelize'
+import 'whatwg-fetch'
+
+export const OAUTH_TOKEN  = 'linksf::oauthToken'
+export const CURRENT_USER = 'linksf::currentUser'
+
+export const authenticate = () => {
+  const user = currentUser()
+
+  if (!user) {
+    window.location = '/login'
+    return false
+  }
+
+  const tokenExpiresAt = new Date(user.tokenExpiresAt*1000)
+  const now = new Date()
+
+  if (tokenExpiresAt < now) {
+    // need to verify token with google
+    const token = sessionStorage[OAUTH_TOKEN]
+    validateGoogleOauthToken(token)
+  } else {
+    // still have a valid token
+    return true
+  }
+}
+
+export const currentUser = () => {
+  try {
+    const user = JSON.parse(sessionStorage[CURRENT_USER])
+
+    if (typeof user == 'object') {
+      return user
+    }
+  } catch (e) {}
```

Comment:
Maybe return `null` here? `undefined` in JS is usually "I forgot to think of this case".